extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-plugin-service-impl.ts

```diff
@@ -0,0 +1,739 @@
+    if (!component.attributes['che-theia.eclipse.org/vscode-preferences']) {
+      component.attributes['che-theia.eclipse.org/vscode-preferences'] = {};
+    }
+
+    // ensure component -> container -> args
+    if (!component.container.args) {
+      component.container.args = ['sh', '-c', '${PLUGIN_REMOTE_ENDPOINT_EXECUTABLE}'];
+    }
+
+    // ensure component -> container -> env
+    if (!component.container.env) {
+      component.container.env = [];
+    }
+
+    // ensure component -> container -> env :: { name: 'PLUGIN_REMOTE_ENDPOINT_EXECUTABLE', value: '...' }
+    const pluginRemoteEndpointExecutable = component.container.env.find(
+      value => value.name === 'PLUGIN_REMOTE_ENDPOINT_EXECUTABLE'
+    );
+    if (!pluginRemoteEndpointExecutable) {
+      component.container.env.push({
+        name: 'PLUGIN_REMOTE_ENDPOINT_EXECUTABLE',
+        value: '/remote-endpoint/plugin-remote-endpoint',
+      });
+    }
+
+    // ensure component -> container -> env :: { name: 'THEIA_PLUGINS', value: '...' }
+    const theiaPlugins = component.container.env.find(value => value.name === 'THEIA_PLUGINS');
+    if (!theiaPlugins) {
+      component.container.env.push({
+        name: 'THEIA_PLUGINS',
+        value: `local-dir:///plugins/sidecars/${component.name}`,
+      });
+    }
+
+    // ensure component -> container -> sourceMapping
+    if (!component.container.sourceMapping) {
+      component.container.sourceMapping = '/projects';
+    }
+
+    // ensure component -> container -> volumeMounts
+    if (!component.container.volumeMounts) {
+      component.container.volumeMounts = [];
+    }
+
+    // ensure component -> container -> volumeMounts :: { name: 'remote-endpoint', path: '/remote-endpoint' }
+    const remoteEndpoint = component.container.volumeMounts.find(value => value.name === 'remote-endpoint');
+    if (!remoteEndpoint) {
+      component.container.volumeMounts.push({
+        name: 'remote-endpoint',
+        path: '/remote-endpoint',
+      });
+    }
+
+    // ensure component -> container -> volumeMounts :: { name: 'plugins', path: '/plugins' }
+    const plugins = component.container.volumeMounts.find(value => value.name === 'plugins');
+    if (!plugins) {
+      component.container.volumeMounts.push({
+        name: 'plugins',
+        path: '/plugins',
+      });
+    }
+  }
+
+  async removePlugin(pluginID: string): Promise<void> {
+    const installedPlugins = await this.getInstalledPlugins();
+    const found = installedPlugins.find(value => value === pluginID);
+    if (!found) {
+      throw new Error(`Plugin ${pluginID} is not installed.`);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const metaYamls: { [pluginId: string]: any } = {};
+
+    try {
+      // read meta.yamls of all installed plugins
+      for (const plugin of installedPlugins) {
+        const yaml = await this.getPluginMetaYaml(plugin);
+        metaYamls[plugin] = yaml;
+      }
```

Comment:
metaYaml will be gone, we can't depend on it --> che-theia-plugins.yaml

Reply:
Done